android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/SessionPrioritizationMiddleware.kt

```diff
@@ -54,27 +66,59 @@ class SessionPrioritizationMiddleware : Middleware<BrowserState, BrowserAction>
                     updatePriorityIfNeeded(state)
                 }
             }
+            is ContentAction.CheckForFormDataAction -> {
+                val tab = context.state.findTab(action.tabId)
+                if (action.containsFormData) {
+                    tab?.engineState?.engineSession?.updateSessionPriority(HIGH)
+                    logger.info("Update the tab ${tab?.id} priority to ${HIGH.name}")
+                    clearHighPriority(context, tab!!.id)
```

Comment:
Maybe we could use `.let` to avoid the use of `!!`
```kotlin
                    tab?.let {
                        clearHighPriority(context, tab.id)
                    }
```

Reply:
Good idea, I'll update this.